mm2src/coins/lightning/ln_rpc.rs

```diff
@@ -1,61 +1,70 @@
+#[cfg(not(target_arch = "wasm32"))] use std::convert::TryFrom;
 
-impl FeeEstimator for UtxoStandardCoin {
+#[cfg(not(target_arch = "wasm32"))]
+impl FeeEstimator for PlatformFields {
     // Gets estimated satoshis of fee required per 1000 Weight-Units.
     fn get_est_sat_per_1000_weight(&self, confirmation_target: ConfirmationTarget) -> u32 {
-        let conf = &self.as_ref().conf;
-        // TODO: Maybe default_fee and confirmation targets can be set in coin configs or lightning configs (would require to move lightning config to coin config) instead
+        let platform_coin = &self.platform_coin;
+
         let default_fee = match confirmation_target {
-            // fetch background feerate
-            ConfirmationTarget::Background => 253,
-            // fetch normal feerate (~6 blocks)
-            ConfirmationTarget::Normal => 2000,
-            // fetch high priority feerate
-            ConfirmationTarget::HighPriority => 5000,
+            ConfirmationTarget::Background => self.default_fees_and_confirmations.background.default_feerate,
+            ConfirmationTarget::Normal => self.default_fees_and_confirmations.normal.default_feerate,
+            ConfirmationTarget::HighPriority => self.default_fees_and_confirmations.high_priority.default_feerate,
         } * 4;
 
+        let conf = &platform_coin.as_ref().conf;
         let n_blocks = match confirmation_target {
-            // fetch background feerate
-            ConfirmationTarget::Background => 12,
-            // fetch normal feerate (~6 blocks)
-            ConfirmationTarget::Normal => 6,
-            // fetch high priority feerate
-            ConfirmationTarget::HighPriority => 1,
+            ConfirmationTarget::Background => self.default_fees_and_confirmations.background.n_blocks,
+            ConfirmationTarget::Normal => self.default_fees_and_confirmations.normal.n_blocks,
+            ConfirmationTarget::HighPriority => self.default_fees_and_confirmations.high_priority.n_blocks,
         };
-        let fee_per_kb = block_on(
-            self.as_ref()
+        let fee_per_kb = tokio::task::block_in_place(move || {
+            platform_coin
+                .as_ref()
                 .rpc_client
                 .estimate_fee_sat(
-                    self.decimals(),
-                    &EstimateFeeMethod::SmartFee,
+                    platform_coin.decimals(),
+                    // Todo: when implementing Native client detect_fee_method should be used for Native and
+                    // EstimateFeeMethod::Standard for Electrum
+                    &EstimateFeeMethod::Standard,
                     &conf.estimate_fee_mode,
                     n_blocks,
                 )
-                .compat(),
-        )
-        .unwrap_or(default_fee);
-        (fee_per_kb as f64 / 4.0).ceil() as u32
+                .wait()
+                .unwrap_or(default_fee)
+        });
+        cmp::max((fee_per_kb as f64 / 4.0).ceil() as u32, 253)
```

Comment:
Could you please add a constant and, if possible, comment on why this is at least 253?

Reply:
Done